services/src/main/java/org/keycloak/protocol/oidc/mappers/ClaimsParameterWithValueIdTokenMapper.java

```diff
@@ -0,0 +1,136 @@
+import org.keycloak.models.ProtocolMapperModel;
+import org.keycloak.models.UserSessionModel;
+import org.keycloak.protocol.oidc.OIDCLoginProtocol;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.representations.ClaimsRepresentation;
+import org.keycloak.representations.IDToken;
+import org.keycloak.util.JsonSerialization;
+import org.keycloak.util.TokenUtil;
+
+/**
+ * @author <a href="mailto:takashi.norimatsu.ws@hitachi.com">Takashi Norimatsu</a>
+ */
+public class ClaimsParameterWithValueIdTokenMapper extends AbstractOIDCProtocolMapper implements OIDCIDTokenMapper {
+
+    public static final String PROVIDER_ID = "oidc-claims-param-value-idtoken-mapper";
+
+    private static final List<ProviderConfigProperty> configProperties = new ArrayList<>();
+
+    public static final String CLAIM_NAME = "claim.name";
+
+    static {
+        ProviderConfigProperty property = new ProviderConfigProperty();
+        property.setName(CLAIM_NAME);
+        property.setLabel("Claim name");
+        property.setType(ProviderConfigProperty.STRING_TYPE);
+        property.setHelpText("Name of the claim you want to set its value. 'true' and 'false can be used for boolean values.");
+        configProperties.add(property);
+
+        OIDCAttributeMapperHelper.addIncludeInTokensConfig(configProperties, ClaimsParameterWithValueIdTokenMapper.class);
+    }
+
+    @Override
+    public String getDisplayCategory() {
+        return TOKEN_MAPPER_CATEGORY;
+    }
+
+    @Override
+    public String getDisplayType() {
+        return "Claims parameter with value ID Token";
+    }
+
+    @Override
+    public String getId() {
+        return PROVIDER_ID;
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Claims specified by Claims parameter with value are put into an ID token.";
+    }
+
+    @Override
+    public List<ProviderConfigProperty> getConfigProperties() {
+        return configProperties;
+    }
+
+    @Override
+    protected void setClaim(IDToken token, ProtocolMapperModel mappingModel, UserSessionModel userSession, KeycloakSession keycloakSession, ClientSessionContext clientSessionCtx) {
+        String claims = clientSessionCtx.getClientSession().getNote(OIDCLoginProtocol.CLAIMS_PARAM);
+        if (claims == null) return;
+
+        if (TokenUtil.TOKEN_TYPE_ID.equals(token.getType())) {
+            putClaims(ClaimsRepresentation.ClaimContext.ID_TOKEN, claims, token, mappingModel, userSession);
+        }
+    }
+
+    private void putClaims(ClaimsRepresentation.ClaimContext tokenType, String claims, IDToken token, ProtocolMapperModel mappingModel, UserSessionModel userSession) {
+        String claimName = mappingModel.getConfig().get(CLAIM_NAME);
+        if (claimName == null) return;
+
+        ClaimsRepresentation claimsRep = null;
+
+        try {
+            claimsRep = JsonSerialization.readValue(claims, ClaimsRepresentation.class);
+        } catch (IOException e) {
+            return;
```

Comment:
Is it possible some warning (or debug message) in the log? Just for the debugging purposes, so the caller is aware what is going on?
As from the opposite perspective of the application developer, who created slightly incorrectly formatted `claims` JSON and sent it to KC, I know that these issues are sometimes hard to debug when something doesn't work and you don't have clue why and there is nothing in server.log :-)

Maybe something same like for example `AcrUtils.getAcrValues ` ? Sorry I did not mention this before.

Reply:
Yes, I will add `LOGGER.warn("Invalid claims parameter", e);` when catching `IOException`.